r-data-manipulation.Rmd

```diff
@@ -0,0 +1,1111 @@
+```{r}
+dognames <- tribble(     
+      ~name, ~n_dogs,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+dognames
+```
+
+* Use `left_join()` to combine the datasets.
+
+* Why doesn't this join work?  Can you fix it?
+```{r}
+babynames_2 <- tribble(
+      ~name, ~n,
+   "CLAIRE",     5207L,
+     "ZURI",      889L,
+     "SUZY",       20L,
+    "VICKI",       13L,
+  "CHELSEA",      929L
+  )
+dognames_2 <- tribble(     
+      ~name,      ~n,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+left_join(babynames_2, dognames_2)
+```
+
+* Why doesn't this join work?  Can you fix it?
+```{r}
+babynames_3 <- tribble(
+      ~name, ~n_babies,
+   "Claire",     5207L,
+     "Zuri",      889L,
+     "Suzy",       20L,
+    "Vicki",       13L,
+  "Chelsea",      929L
+  )
+dognames_3 <- tribble(     
+      ~name,      ~n_dogs,
+  "CHELSEA",      1L,
+   "CLAIRE",      2L,
+     "SUZY",      3L,
+    "VICKI",      1L,
+     "ZURI",      1L
+  )
+left_join(babynames_3, dognames_3)
+```
+
+
+### Exercise
+
+Find the dog licences per capita, but now at the community district level, by joining the `dog_licenses` data with the `nyc-cd-pop` data.
+
+## How can I read my own tabular data into R?
+
+### What is tabular data?
+
+Tabular data describes data that is in the form of a table: values arranged in rows each of the same length, or equivalently values arranged in columns each of the same length. Here's a small example of some tabular data:
+```{r, echo = FALSE, message=FALSE}
+example_csv <- here("data", "site.csv")
+sites <- read_csv(example_csv )
+knitr::kable(sites)
+```
+
+Each row records information on a site at which water measurements are taken.  There are three columns: a site identification code, and the location of the site in latitude and longitude. 
+
+This is an incredibly common way of *displaying* data, but when *storing* tabular data in a file, we need a way to communicate when records and values begin and end.  A very popular format for doing this is CSV.  CSV, is short for **c**omma **s**eparated **v**alues, and like the name suggests, a comma, `,`, is used to separate the values for each column, while each record goes on a new line.  The file is plain text, but we use the extension `.csv` to indicate that is follows the CSV format conventions.
+
+Here's how the table above would look inside the CSV file `site.csv`:
+```{r, echo = FALSE, comment = NA}
+writeLines(readLines(example_csv))
+```
+
+In this case the first line has the column names, this is common (and recommended!), but not universal. 
+
+Why is CSV so popular? 
+
+* It's human readable.  CSV isn't a special file type, it is just a plain text file that follows some conventions.  This means you don't need any special software to look at the contents -- you can open in up in anything that can examine text and take a look inside. 
```

Comment:
```suggestion
* It's human readable.  CSV isn't a special file type, it is a simple plain text file that follows some conventions.  This means you don't need any special software to look at the contents -- you can open it up in anything that can examine text and take a look inside. 
```

Reply:
Good catch!  This is one of my bad habits...